### PR TITLE
WaveTournament | Custom Tournament Client Design using gosumemory

### PR DESCRIPTION
**Pull request by VictimCrasher**
A reference design for gosumemory tournament support
Initially want to keep it by myself but it might be helpful for others to experiment

### In action:
[https://twitter.com/Victim_Crasher/status/1319851426600767488](https://twitter.com/Victim_Crasher/status/1319851426600767488)

### Features:
- 1v1 to 4v4 OK
- Best of Score automatically if you change it on tournament client 
- ScoreV2 OK (thanks for fix BlackShark)
- Team Name OK (thanks for fix BlackShark)
- Score calculation should have no bug now due gosumemory taking game score directly from client (again, thanks for fix BlackShark)
- (Should have been) Easy to customize, just edit the pngs on `WaveTournament/static` folder

## Screenshots

-----

### To code reviewer:
Please check `index.js` line 97, and please point me if there's more efficient way to change background image other than
```javascript
if(tempImg !== data.menu.bm.path.full){
    tempImg = data.menu.bm.path.full;
    data.menu.bm.path.full = data.menu.bm.path.full.replace(/#/g,'%23').replace(/%/g,'%25').replace(/\\/g,'/');
    mapContainer.style.backgroundImage = `url('http://127.0.0.1:24050/Songs/${data.menu.bm.path.full}?a=${Math.random(10000)}')`;
}
```
Because sometimes it causes error and the image won't load

### CODE REVIEW

File: WaveTournament/index.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+let scoreRedTemp;
+let teamNameBlueTemp;
+let teamNameRedTemp;
+let gameState;
+
+socket.onmessage = event => {
+    let data = JSON.parse(event.data);
+	if(teamSizeTemp !== data.tourney.ipcClients.length) {
+		teamSizeTemp = data.tourney.ipcClients.length;
+	}
```
Comment (Dartandr) [inline WaveTournament/index.js]: unused variable Xd actually not sure why it exist

**Comment by Dartandr** [comment]
About image loading, can you give some examples otherwise i don't see any problems

File: WaveTournament/index.js (hunk 1 above)
Comment (VictimCrasher) [reply WaveTournament/index.js]: It was previously used for counting the score manually, it seems that I forgot to delete that one, fix in a second.

**Comment by VictimCrasher** [comment]
> About image loading, can you give some examples otherwise i don't see any problems

I did this because of windows file system that using `\` instead of `/`
Might not be the best if used in other OS but then again BlackShark said that it wont support other OS atm.

```javascript
.replace(/\\/g,'/');
```

**Comment by Dartandr** [comment]
> 
> 
> > About image loading, can you give some examples otherwise i don't see any problems
> 
> I did this because of windows file system that using `\` instead of `/`
> Might not be the best if used in other OS but then again BlackShark said that it wont support other OS atm.
> 
> ```js
> .replace(/\\/g,'/');
> ```

well i figured out you need to use this
```js
.replace(/\\/g,'/');
```
because you changing style of object in other counters i preferred to use src instead, it's okay i guess just web dev things 
https://github.com/l3lackShark/static/blob/fb959c3f09e1fa6715c6b62f2949a7683b5e5612/Classic/index.js#L62

**Comment by VictimCrasher** [comment]
> > > About image loading, can you give some examples otherwise i don't see any problems
> > 
> > 
> > I did this because of windows file system that using `\` instead of `/`
> > Might not be the best if used in other OS but then again BlackShark said that it wont support other OS atm.
> > ```js
> > .replace(/\\/g,'/');
> > ```
> 
> well i figured out you need to use this
> 
> ```js
> .replace(/\\/g,'/');
> ```
> 
> because you changing style of object in other counters i preferred to use src instead, it's okay i guess just web dev things
> https://github.com/l3lackShark/static/blob/fb959c3f09e1fa6715c6b62f2949a7683b5e5612/Classic/index.js#L62

Yea, I'm using different approaches. If it's still acceptable than I'm okay leaving it as is.

**Comment by l3lackShark** [comment]
Thanks!

**Pull request merged**